simulator/virtual_machine_test.go

```diff
@@ -390,6 +390,104 @@ func TestReconfigVmDevice(t *testing.T) {
 	}
 }
 
+func TestConnectVmDevice(t *testing.T) {
+	ctx := context.Background()
+
+	m := ESX()
```

Comment:
Note for future reference, you can use:
```go
Test(func() { ... }, ESX())
```
Takes care of vcsim startup/teardown and creating the client (lines 396..409 here).
See: https://github.com/vmware/govmomi/blob/586f608c55bc815289af45963a7c9bed63138af0/simulator/model.go#L912-L913